src/core/redux/reducers/resetRequestState.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * @flow
+ */
+
+import { Map } from 'immutable';
+import { LangUtils } from 'lattice-utils';
+
+import { RS_INITIAL_STATE } from '../constants';
+import type { ResetRequestStatesAction } from '../actions';
+
+const { isNonEmptyArray } = LangUtils;
+
+export default function reducer(state :Map, action :ResetRequestStatesAction) {
+
+  const { requestStateAction = [] } = action;
+  let newState = state;
+  if (isNonEmptyArray(requestStateAction) && state.hasIn(requestStateAction)) {
+    newState = newState.setIn(requestStateAction, RS_INITIAL_STATE);
+  }
+
+  return newState;
+}
```

Comment:
Same comment. I don't think this reducer needs to exist since `resetRequestStates` exists.

Also, the flow annotation is incorrect. It should be `ResetRequestStateAction` (singular).